### PR TITLE
Update `dusk-plonk` dependency to `plonkup`

### LINKED ISSUE #80: Update plonkup dependency to `plonkup` `v0.1.0`
**Describe what you want implemented** 
Update the `dusk_plonk` dependency to `plonkup` `v0.1.0`

**Describe "Why" this is needed**
For the versions `v0.9` and `v0.10` the `dusk-plonk` repo actually reflected the PlonkUP protocol and not PLONK. After moving PlonkUP to its own repo, the crates using PlonkUP need to be updated accordingly. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by moCello**

### CODE REVIEW

File: CHANGELOG.md
Comment (ZER0) [inline CHANGELOG.md]: Nit: It should start with either `Change` or one of the aliases we normally use (e.g. `Rename`, `Update`…)

File: benches/merkle.rs
Comment (ZER0) [inline benches/merkle.rs]: I think we probably need to rename this `PlonkupError` as well.

File: src/components/bars/compose.rs
Comment (ZER0) [inline src/components/bars/compose.rs:93]: You can also use shadowing when you're not using anymore a variable.
Having something like this:
```rust
        let expected = [
            678, 674, 683, 687, 690, 660, 689, 686, 692, 678, 661, 668, 686,
            662, 680, 666, 672, 684, 669, 683, 687, 682, 674, 663, 673, 660,
            656,
        ];
        
        let expected = super::native(expected);
```

is perfectly idiomatic.

File: src/components/bars/compose.rs
Comment (ZER0) [inline src/components/bars/compose.rs]: Remember that `assert*` macro also takes an additional parameter that works as "context"; it's useful because it give context to both the reader / maintainer and in the logging.

File: src/components/concrete.rs
Comment (ZER0) [inline src/components/concrete.rs:71]: Are these actually unit test? E.g. are they accessing to internals of the module (in this case `concrete`) that are not exposed to the library's consumer?

File: src/lib.rs
Comment (ZER0) [inline src/lib.rs]: I think this is either a left-over or a mistake.

File: tests/merkle.rs
Comment (ZER0) [inline tests/merkle.rs]: Ditto.

File: tests/merkle.rs
Comment (ZER0) [inline tests/merkle.rs]: I don't think you need `deref()` here.

**Review by ZER0 — CHANGES_REQUESTED** [review]
Overall looks good, there is few nits; but mostly I mark as request change for two main reason:
- The `alloc` commented out
- The bench are not compiling at all (is that known?)

As separate note, looks like you're not locally running clippy, since I noticed a lot of small things that clippy would have definitely catch; but I didn't report them, since are not strictly related to the PR. My suggestion would be open an issue to add `clippy` to the CI, as we have in other repo, and fixes all the nits that make sense (or `allow` the one they doesn't).

File: src/components/bars/compose.rs
Comment (moCello) [reply src/components/bars/compose.rs:93]: That is true, but is anything gained in the above case? I would argue that here it is adding confusion since the array is not the expected value, but its decomposition.

File: src/components/concrete.rs
Comment (moCello) [reply src/components/concrete.rs:71]: Yes, everything in the `components` module is only public within the crate and needs to be tested internally

File: src/components/bars/compose.rs
Comment (moCello) [reply src/components/bars/compose.rs]: What would you suggest for context?

File: tests/merkle.rs
Comment (ZER0) [inline tests/merkle.rs]: While there is nothing technically wrong with that, I would rather derive `Default` for `MerkleOpeningCircuit` since all its member implement `Default`, and having here just:
```rust
let mut circuit = MerkleOpeningCircuit::default();
```

File: src/components/bars/compose.rs
Comment (ZER0) [reply src/components/bars/compose.rs:93]: It gains a reduction of pollution of different variables, and a clear statement (for the developer) that from that point on the previous one `composition` is not used anymore. Shadowing can be problematic when they're across pages of code, but if they're don in the same visual block there is no confusion in that sense, it actually helps.

File: src/components/bars/compose.rs
Comment (ZER0) [reply src/components/bars/compose.rs]: I would add what you're asserting here. There is no context, however you added a comment above if I'm not mistaken saying `// Check if -5 is composed correctly`. That is the context, whatever this assert is here for.

**Review by ZER0 — APPROVED** [review]
LGTM there a very minor nit about the default, but since it's just for testing it's up to you if you want to fix it before merging (I would do, you basically have to add only a `#derive(Default)` and remove few lines of code, that is always a good thing, improving readability in the meantime).

**Pull request merged**
